src/Microsoft.DotNet.Cli.Utils/Product.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Reflection;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class Product
+    {
+        public static readonly string LongName = ".NET Command Line Tools";
+        public static readonly string Version = GetProductVersion();
+
+        private static string GetProductVersion()
+        {
+            var attr = typeof(Product).GetTypeInfo().Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>();
```

Comment:
> If I had a tool that needed username/password on the cmd line, would it get picked up?

Ideally, the user's command-line arguments shouldn't leak off the box in a recoverable form.

A pattern that has worked well elsewhere is to one-way hash potentially sensitive data before sending it up to a telemetry system.

In this case, the commandline could be parsed into its separate terms, and each term could be oneway hashed (with sha256, for example), and only the hashes should be sent up, ideally in the order that they were specified on the commandline.

This approach allows data analysts to understand how invocations vary based on their parameters, without being able to determine what the parameter names or values are. 

It also means that publicly known standard parameters can be "unhashed", because you can compare observed hashes in the data to the hashes of strings you independently know.  

Contrastingly, custom parameter names, and obviously, parameter values like email address and passwords, will not ever leave the customer's machine, and, unless these are guessable, they cannot be recovered via hash comparison.